src/generate-types.ts

```diff
@@ -352,6 +352,26 @@ export function run(options: Options) {
     };
   }
 
+  function generateHeadersSchemaType(headers: oas.HeadersObject): oas.SchemaObject {
+    const required: string[] = [];
+    const properties = Object.entries(headers).reduce((memo, [headerName, headerObject]) => {
+      if (oautil.isReferenceObject(headerObject)) {
+        required.push(headerName);
+        return { ...memo, [headerName]: headerObject };
+      } else if (headerObject.schema) {
+        required.push(headerName);
+        return { ...memo, [headerName]: headerObject.schema };
+      }
+      return memo;
+    }, {} as { [key: string]: oas.SchemaObject | oas.ReferenceObject });
+    return {
+      type: 'object',
+      properties,
+      required,
+      additionalProperties: true
```

Comment:
`true` so that we don't need to declare *all*  the headers

Reply:
we drop extra headers for requests in server side would that work here or are we keen to allow also unknown headers?

Reply:
additionalproperties are a bit me in general as those get turned into `{[key]: unknown}` which are not really nice to use.